hardware_reservations.go

```diff
@@ -0,0 +1,100 @@
+package packngo
+
+import "fmt"
+
+const hardwareReservationBasePath = "/hardware-reservations"
+
+// HardwareReservationService interface defines available hardware reservation functions
+type HardwareReservationService interface {
+	Get(hardwareReservationID string, listOpt *ListOptions) (*HardwareReservation, *Response, error)
+	List(projectID string, listOpt *ListOptions) ([]HardwareReservation, *Response, error)
+	Move(string, string) (*HardwareReservation, *Response, error)
+}
+
+// HardwareReservationServiceOp implements HardwareReservationService
+type HardwareReservationServiceOp struct {
+	client *Client
+}
+
+// HardwareReservation struct
+type HardwareReservation struct {
+	ID        string    `json:"id,omitempty"`
+	ShortID   string    `json:"short_id,omitempty"`
+	Facility  Facility  `json:"facility,omitempty"`
+	Plan      Plan      `json:"plan,omitempty"`
+	Href      string    `json:"href,omitempty"`
+	Project   Project   `json:"project,omitempty"`
+	Device    Device    `json:"device,omitempty"`
```

Comment:
I would do the Device a pointer. It's then easier to test if the reservation is occupied.